python/cuml/feature_extraction/text.py

```diff
@@ -102,34 +104,39 @@ def _get_stop_words(self):
         else:  # assume it's a collection
             return list(self.stop_words)
 
-    def get_char_ngrams(self, n, str_series, doc_id_sr):
+    def get_char_ngrams(self, ngram_size, str_series, doc_id_sr):
         """
         Handles ngram generation for characters analyzers.
 
         When analyzer is 'char_wb', we generate ngrams within word boundaries,
         meaning we need to first tokenize and pad each token with a delimiter.
         """
-        if self.analyzer == 'char':
-            token_count = str_series.str.len()
-            ngram_sr = str_series.str.character_ngrams(n=n)
-            ngram_count = None
-        else:
-            # here analyzer is 'char_wb'
+        if self.analyzer == 'char_wb' and ngram_size != 1:
             tokens = str_series.str.tokenize(self.delimiter)
 
             padding = Series(self.delimiter).repeat(len(tokens))
             tokens = padding.str.cat(tokens.str.cat(padding))
             tokens = tokens.reset_index(drop=True)
 
-            ngram_sr = tokens.str.character_ngrams(n=n)
+            ngram_sr = tokens.str.character_ngrams(n=ngram_size)
 
             token_count = str_series.str.token_count(self.delimiter)
             doc_id_df = cudf.DataFrame({
                 'doc_id': doc_id_sr.repeat(token_count).reset_index(drop=True),
                 # formula to count ngrams given number of letters per token:
-                'ngram_count': tokens.str.len() - (n - 1)
+                'ngram_count': tokens.str.len() - (ngram_size - 1)
```

Comment:
should get_ngrams below also be updated to take `ngram_size` rather than n? (I think ngram_size is nice and clear)

Reply:
Agreed.